src/gui/editorwidgets/qgsrelationreferencesearchwidgetwrapper.cpp

```diff
@@ -145,23 +145,8 @@ void QgsRelationReferenceSearchWidgetWrapper::onValueChanged( const QVariant &va
 
 void QgsRelationReferenceSearchWidgetWrapper::setExpression( QString exp )
 {
-  QgsSettings settings;
-  QString nullValue = QgsApplication::nullRepresentation();
   QString fieldName = layer()->fields().at( mFieldIdx ).name();
-
-  QString str;
-  if ( exp == nullValue )
```

Comment:
Actually this logic looks a bit different - seems to allow entry of "null" string in the search widget wrapper to search for null values. Better not change this one.